scalding-hadoop-test/src/main/scala/com/twitter/scalding/platform/HadoopPlatform.scala

```diff
@@ -33,6 +35,9 @@ trait HadoopPlatform[P, R, T <: HadoopPlatform[P, R, T]] {
 
   def run(): Unit
 
+  def runAndHandle(fn: Throwable => Unit): Unit =
+    fn(Try { run() }.failed.get)
```

Comment:
wait, is this what you expect? If the job succeeds, `.failed` still gives an exception. Can you comment then intent of this method? Interestingly, here is a case where adding a type parameter helps explain.

If you wrote:
```scala
def runAndHandle[T](fn: Throwable => T): T =
```
we could see in the types that `fn` must be called, so even if the job succeeds, an exception will still be generated and passed to `fn`. But since anyone can create a value of `Unit`, I can't tell if you meant:
```scala
Try(run()) match {
  case Failure(err) => fn(err)
  case Success(_) => ()
}
```

If you really mean what you have, we should probably call it `def runExpectFailure` or something.